content/geometry/CircleCircleArea.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * Author: Takanori MAEHARA, chilli
+ * Date: 2019-11-03
+ * License: CC0
+ * Source: https://github.com/spaghetti-source/algorithm/blob/master/geometry/_geom.cc#L729
+ * Description: Calculates the area of the intersection of 2 circles
+ * Status:
+ */
+
+template<class P>
+double circleCircleArea(P c, double cr, P d, double dr) {
+	if (cr < dr) swap(c, d), swap(cr, dr);
```

Comment:
hm, my suggestion to add `swap(c, d)` was silly, only the distance between them is used and that's symmetric